lib/babe/epoch_handler.go

```diff
@@ -44,23 +44,57 @@ func newEpochHandler(epochNumber, firstSlot uint64, epochData *epochData, consta
 			epochData.threshold,
 			keypair,
 		)
-		if errors.Is(err, errOverPrimarySlotThreshold) {
+
+		switch err {
+		case nil:
```

Comment:
Why do we use a switch here instead of the usual `if err != nil` or `err == nil`? That would bring less indentation

Reply:
addressed